wgpu-core/src/device/mod.rs

```diff
@@ -2528,10 +2529,11 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         let (cmdbuf, _) = hub
             .command_buffers
             .unregister(command_encoder_id, &mut token);
-
-        let device = &mut device_guard[cmdbuf.device_id.value];
-        device.untrack::<G>(&hub, &cmdbuf.trackers, &mut token);
-        device.cmd_allocator.discard(cmdbuf);
+        if let Some(cmdbuf) = cmdbuf {
```

Comment:
I could not combine the `if let` statement with the function call here (and some more places) because of the mutable borrow of `token` in `unregister()` call.

Reply:
not sure what you wanted to do. Could you clarify?

Reply:
Similar to what you suggested in the previous review, i.e.
```rust
if let (Some(cmdbuf), _) = hub
            .command_buffers
            .unregister(command_encoder_id, &mut token) { ... }
```

Reply:
ah I see. Thanks for clarifying!